internal/provider/resource_consul_cluster.go

```diff
@@ -261,6 +272,29 @@ func resourceConsulClusterCreate(ctx context.Context, d *schema.ResourceData, me
 		return diag.Errorf("specified Consul version (%s) is unavailable; must be one of: %v", consulVersion, availableConsulVersions)
 	}
 
+	// If specified, validate and parse the primary link provided for federation.
+	primary_link, ok := d.GetOk("primary_link")
+	var primary *sharedmodels.HashicorpCloudLocationLink
+	if ok {
+		primary, err = parseLinkURL(primary_link.(string), ConsulClusterResourceType)
+		if err != nil {
+			return diag.Errorf(err.Error())
+		}
+		if primary.Location.OrganizationID == "" {
+			primaryOrgID, err := clients.GetParentOrganizationIDByProjectID(ctx, client, primary.Location.ProjectID)
+			if err != nil {
+				return diag.Errorf("Error determining organization of primary cluster. %v", err)
+			}
+			primary.Location.OrganizationID = primaryOrgID
+		}
+		// fetch the primary cluster
+		primaryConsulCluster, err := clients.GetConsulClusterByID(ctx, client, primary.Location, primary.ID)
+		if err != nil {
+			return diag.Errorf("unable to check for presence of an existing primary Consul cluster (%s): %v", primary.ID, err)
+		}
+		primary.Location.Region = primaryConsulCluster.Location.Region
+	}
```

Comment:
💭  The provider does some basic validation, like input format and verifying the existence of dependent resources, but I'd agree that this seems closer to Consul business logic. That being said, I could see the advantage of having the request fail sooner, giving the user quicker feedback. How much later down the line would the consul service check for the primary?

Reply:
Let me verify if this is done in the consul-service. If not, definitely sounds like a good place to include it there vs here.

Reply:
Nevermind. The reason I have this check in here is because I need the primary's region config (primary.Location.Region). Else, the below error is raised:
```hcp_consul_cluster.secondary: Creating...

Error: unable to create Consul cluster (consul-cluster-secondary): [POST /consul/2020-08-26/organizations/{cluster.location.organization_id}/projects/{cluster.location.project_id}/clusters][400] ConsulService_Create default  &{Code:3 Details:[0xc000151e60] Error:config: (consul_config: (primary: (location: (region: cannot be blank.).).).). Message:config: (consul_config: (primary: (location: (region: cannot be blank.).).).).}```